### DISCUSSION TITLE
Refactor page transformations

### DISCUSSION THREAD

**Discussion by MartinThoma** [py-pdf/pypdf#851] [Ideas]
I would like to deprecate those [`Page`](https://pypdf2.readthedocs.io/en/latest/modules/PageObject.html) methods:

1. `mergeTransformedPage(self, page2, ctm, expand=False)`
2. `mergeScaledPage(self, page2, scale, expand=False)`
3. `mergeRotatedPage(self, page2, rotation, expand=False)`
4. `mergeRotatedTranslatedPage(self, page2, rotation, tx, ty, expand=False)`
5. `mergeRotatedTranslatedPage(self, page2, rotation, tx, ty, expand=False)`
6. `mergeRotatedScaledPage(self, page2, rotation, scale, expand=False)`
7. `mergeScaledTranslatedPage(self, page2, scale, tx, ty, expand=False)`
8. `mergeRotatedScaledTranslatedPage(self, page2, rotation, scale, tx, ty, expand=False)`

Instead of those, I would like the users to use a combination of

* `mergePage(self, page2)`: Ensure that this does not loose transformations of page2 + add `expand` parameter
* `translate(self)`: Doesn't exist yet
* `scale(self, sx, sy)` / `scaleBy(self, factor)`
* `rotateClockwise(self, angle)`

The idea would be to have a smaller + better tested interface without loosing any functionality. The additional effort on the users side should be minimal.

How do you think / feel about this?

**Comment by johns1c** [comment]
Looks a good idea as will simplify things.

I have not looked at the code but can the translate, scale and rotate just be "ease of use" routines just calling a single transform functon that carries out a standard Afine transformation.   


Chris

**Reply by MartinThoma** [reply]
I just merged https://github.com/py-pdf/PyPDF2/pull/878 yesterday which adds the `Transformation` object. It allows the user to simply represent a set of transformations in one object. That would make the other function unnecessary.

**Reply by MartinThoma** [reply]
It doesn't quite do what I originally thought about, but that might just be a small addition (essentially storing a Transformation object as an attribute of a Page)
